public/Students/Leo Jayachandran/DynamicIndex/dynamicIndexMaker.js

```diff
@@ -1,13 +1,15 @@
 (function () {
 	processContents();
-	let indexExclusions = ["source", 'lulu'];
-	let indexRequirements = "";
+	let indexExclusions = ["source", 'lulu' , "@"]; // Case insensitive
+	let indexRequirements =  [""]; // Case sensisitve  if ends with @, that tag will be trimmed, else it will be left as it(Can be entered here or in currentScript.dataset.filter)
 	
 	
 	//main function
 	async function processContents() {
-		$(document.currentScript).after(`<div id="indexDiv"><p id="indexLetterList"></p><div id="indexTable"></div></div>`);
-		
+        $(document.currentScript).after(`<div id="indexDiv"><p id="indexLetterList"></p><div id="indexTable"></div></div>`);
+        if (typeof(currentScript) !== "undefined" && typeof(currentScript.dataset) !== "undefined" && typeof(currentScript.dataset.filter) !== "undefined") {
```

Comment:
`currentScript` is undefined here. Either use `document.currentScript` or copy the contents of `document.currentScript` into a new variable called `currentScript`.